### PR TITLE
chore: introduce pytorch tensor instead of pytorch wav

### PR DESCRIPTION
**Pull request by bolgheroni**

### CODE REVIEW

File: src/spira_training/shared/adapters/pytorch/model_trainer/implementations/pytorch_dataset.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+    PytorchTensor,
 )
-from src.spira_training.shared.core.models.dataset import Dataset
 import torch.utils.data
 
 
-class PytorchDataset(torch.utils.data.Dataset, Dataset):
-    def __init__(self, features: List[PytorchWav], labels: list[int]):
+class PytorchDataset(torch.utils.data.Dataset):
+    def __init__(self, features: List[PytorchTensor], labels: List[PytorchLabel]):
```
Comment (lucasqml) [inline src/spira_training/shared/adapters/pytorch/model_trainer/implementations/pytorch_dataset.py:12]: já que aqui colocou o PytorchLabel pra label, será que não seria bom colocar um PytorchFeature pra feature? que ai fica semanticamente mais concordante os dois

Comment (bolgheroni) [reply src/spira_training/shared/adapters/pytorch/model_trainer/implementations/pytorch_dataset.py:12]: acho que fica mais confuso... prefiro assim

File: src/spira_training/shared/adapters/pytorch/model_trainer/implementations/basic_pytorch_audio_factory.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from src.spira_training.shared.core.models.audio import Audio
+import torch
+from src.spira_training.shared.adapters.pytorch.model_trainer.interfaces.pytorch_audio_factory import (
+    PytorchTensorFactory,
+)
+
+
+class BasicPytorchTensorFactory(PytorchTensorFactory):
+    def create_tensor_from_audio(self, audio: Audio) -> PytorchTensor:
+        return PytorchTensor(torch.tensor(audio.wav.tensor, dtype=torch.float32))
```
Comment (lucasqml) [inline src/spira_training/shared/adapters/pytorch/model_trainer/implementations/basic_pytorch_audio_factory.py:13]: eu achei que dps da nossa conversa a gnt ia só usar o audio.wav.tensor, assumindo que isso já era um tensor dentro do nosso código

Comment (lucasqml) [reply src/spira_training/shared/adapters/pytorch/model_trainer/implementations/basic_pytorch_audio_factory.py:13]: já tinha até removido do meu código pro novo PR as referencias a factory que constroi a partir de um Audio

Comment (bolgheroni) [reply src/spira_training/shared/adapters/pytorch/model_trainer/implementations/basic_pytorch_audio_factory.py:13]: so mantive pq vc tava usando
